package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,537 @@
+        """number of frames in trajectory"""
+        for name, value in self._has.items():
+            if value:
+                return self._particle_group[name]['value'].shape[0]
+
+    def _reopen(self):
+        """reopen trajectory"""
+        if self.parallel:
+            self._frame = -1
+            self._read_next_timestep()
+        else:
+            self.close()
+            self.open_trajectory()
+
+    def _read_frame(self, frame):
+        """reads data from h5md file and copies to current timestep"""
+        try:
+            for name, value in self._has.items():
+                if value:
+                    myframe = self._particle_group[name]['step'][frame]
+                    break
+            else:
+                raise NoDataError("Provide at least a position, velocity"
+                                  " or force group in the h5md file.")
+        except ValueError:
+            raise IOError from None
+
+        self._frame = frame
+        ts = self.ts
+        particle_group = self._particle_group
+        ts.frame = frame
+
+        # this block populates the data dictionary
+        # DT not read
+        if 'observables' in self._file:
+            data = self._file['observables']
+            for name in self._data_keywords:
+                self._copy_data(name, data)
+        for name, value in self._has.items():
+            if value:
+                self.ts.data['time'] = particle_group[name]['time'][frame]
+                break
+
+        # set frame box dimensions
+        # H5MD files must contain 'box' group in each particle group
+        ts._unitcell[:] = particle_group['box/edges/value'][frame, :]
+
+        # set the timestep positions, velocities, and forces with
+        # current frame dataset
+        if self._has['position']:
+            ts.positions = self._get_frame_dataset('position')
+        if self._has['velocity']:
+            ts.velocities = self._get_frame_dataset('velocity')
+        if self._has['force']:
+            ts.forces = self._get_frame_dataset('force')
+
+        # unit conversion
+        if self.convert_units:
+            # converts units if convert_units=True
+            self._convert_units()
+
+        return ts
+
+    def _read_next_timestep(self):
+        """read next frame in trajectory"""
+        return self._read_frame(self._frame + 1)
+
+    def _get_frame_dataset(self, dataset):
+        frame_dataset = self._particle_group[
+                        dataset]['value'][self.frame, :]
+        n_atoms_now = frame_dataset.shape[0]
+        if n_atoms_now != self.n_atoms:
+            raise ValueError("Frame {} has {} atoms but the initial frame"
+                             " has {} atoms. MDAnalysis is unable to deal"
+                             " with variable topology!"
+                             "".format(self.frame,
+                                       n_atoms_now,
+                                       self.n_atoms))
+        else:
```

Comment:
Omit the `else` and dedent the return. Using `if` to terminate early is perfectly fine and having the `return` as the last element of the normal block is good style. It reinforces the idea that the `if` condition is an interruption to the normal flow of control.

And it saves a line of code.